packages/canary-web-components/src/components/ic-data-table/ic-data-table.tsx

```diff
@@ -1691,17 +1703,40 @@ export class DataTable {
           : this.getSortFunction()
       )
       .map((row, index) => {
+        const isRowSelected =
+          this.highlightSelectedRow && this.selectedRows.includes(row);
+        const cellIndex = index + paginationOffset;
+
         return (
           <tr
-            // eslint-disable-next-line react/jsx-no-bind
-            onClick={() => this.onRowClick(row)}
             class={{
-              ["table-row"]: true,
-              ["table-row-selected"]:
-                this.highlightSelectedRow && this.selectedRow === row,
+              "table-row": true,
+              "table-row-selected": isRowSelected,
             }}
           >
-            {this.createCells(row, index + paginationOffset)}
+            {this.highlightSelectedRow && (
+              <td
+                class={{
+                  "table-cell": true,
+                  "checkbox-cell": true,
+                  [`table-density-${this.density}`]: this.notDefaultDensity(),
+                }}
+              >
+                <div class="checkbox-wrapper">
+                  <ic-checkbox
+                    class="ic-data-table-checkbox"
+                    checked={isRowSelected}
+                    disabled={this.updating || this.loading}
+                    hideLabel
+                    label={cellIndex}
```

Comment:
I feel like this label could be changed to be a bit more helpful for screen reader users. Maybe it could be "Select row \<row number\>"? And then it changes to "Deselect..." when selected

(Don't know if including the row number is necessary but might be useful)